src/core/execute.c

```diff
@@ -585,24 +586,24 @@ int exec_spawn(
                                   "--log-level", max_log_levels,
                                   "--log-target", log_target_to_string(manager_get_executor_log_target(unit->manager))),
                         environ,
-                        cg_unified() > 0 ? subcgroup_path : NULL,
+                        cg_unified() > 0 ? params->cgroup_path : NULL,
```

Comment:
Well, this is wrong. For control processes we'd start to violate the rule of cgv2 that only leaf cgs shall contain processes...

Reply:
I don't see how this can ever work with control processes. Maybe the way forward is to degrade ProtectControlGroups=private to yes for them and document the behavior.

Reply:
Ah these get spawned in live cgroups? Damn OK yes this will need some rework then